soc/xtensa/intel_adsp/common/printk_out.c

```diff
@@ -24,9 +24,14 @@
 #define SLOT_SIZE 64
 #define SLOT_MAGIC 0x55aa
 
-#define NSLOTS (CONFIG_ADSP_LOG_WIN_SIZE / SLOT_SIZE)
+#define NSLOTS (SRAM_TRACE_SIZE / SLOT_SIZE)
 #define MSGSZ (SLOT_SIZE - sizeof(struct slot_hdr))
 
+/* Translates a SRAM pointer into an address of the same memory in the
+ * uncached region from 0x80000000-0x9fffffff
+ */
+#define UNCACHED_PTR(p) ((void*)(((int)p) & ~0x20000000))
```

Comment:
We have this in a header, but we do need to de-duplicate the headers too. Lets use the header versions when we do the cleanup.